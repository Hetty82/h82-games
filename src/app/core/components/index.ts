import { AppComponent } from './app/app.component'
import { LayoutComponent } from './layout/layout.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const components: any[] = [
  AppComponent,
  LayoutComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './layout/layout.component'
export * from './not-found/not-found.component'
