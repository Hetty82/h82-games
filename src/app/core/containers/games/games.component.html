<h2>Game list</h2>

<ul>
  <li *ngFor="let game of (games$ | async)">{{ game.name }}</li>
</ul>
