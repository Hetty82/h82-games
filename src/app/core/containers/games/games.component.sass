@import '~styles/variables'

.item
  margin-bottom: $spacing-sm
