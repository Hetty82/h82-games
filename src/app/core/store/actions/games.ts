import { Action } from '@ngrx/store'

import { Game } from '../../models/game.interface'


export type GameError = any
export type GameName = string

// Load games
export const LOAD_GAMES = '[Games] Load Games'
export const LOAD_GAMES_FAIL = '[Games] Load Games Fail'
export const LOAD_GAMES_SUCCESS = '[Games] Load Games Success'
// Select game
export const SELECT_GAME = '[Games] Select Game'


// Actions
export class LoadGames implements Action {
  readonly type = LOAD_GAMES
}

export class LoadGamesFail implements Action {
  readonly type = LOAD_GAMES_FAIL
  constructor(public payload: GameError) {}
}

export class LoadGamesSuccess implements Action {
  readonly type = LOAD_GAMES_SUCCESS
  constructor(public payload: Game[]) {}
}

export class SelectGame implements Action {
  readonly type = SELECT_GAME
  constructor(public payload: GameName) {}
}


// Action types
export type GamesAction =
  | LoadGames
  | LoadGamesFail
  | LoadGamesSuccess
  | SelectGame
