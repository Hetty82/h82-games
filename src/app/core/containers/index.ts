import { AppComponent } from './app/app.component'
import { GamesComponent } from './games/games.component'
import { LoginComponent } from './login/login.component'
import { NotFoundComponent } from './not-found/not-found.component'

export const containers: any[] = [
  AppComponent,
  GamesComponent,
  LoginComponent,
  NotFoundComponent,
]

export * from './app/app.component'
export * from './games/games.component'
export * from './login/login.component'
export * from './not-found/not-found.component'
