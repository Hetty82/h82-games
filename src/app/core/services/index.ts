import { GamesService } from './games.service'
import { UserService } from './user.service'

export const services: any[] = [
  GamesService,
  UserService,
]

export * from './games.service'
export * from './user.service'
