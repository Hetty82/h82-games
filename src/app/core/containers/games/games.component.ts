import { Component, OnInit } from '@angular/core'
import { Observable } from 'rxjs/Observable'

import { GamesService } from '../../services'
import { Game } from '../../interfaces/game.interface'


@Component({
  selector: 'app-games',
  templateUrl: './games.component.html',
  styleUrls: ['./games.component.sass']
})
export class GamesComponent implements OnInit {
  games$: Observable<Game[]> = this.gamesService.getGames()

  constructor(
    private gamesService: GamesService
  ) { }

  ngOnInit() {
  }
}
