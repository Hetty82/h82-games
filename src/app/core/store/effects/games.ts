import { Injectable } from '@angular/core'

import { Effect, Actions } from '@ngrx/effects'
import { of } from 'rxjs/observable/of'
import { map, switchMap, catchError } from 'rxjs/operators'

import * as gamesActions from '../actions/games'
import * as fromServices from '../../services'


@Injectable()
export class GamesEffects {
  constructor(
    private actions$: Actions,
    private gamesService: fromServices.GamesService
  ) {}

  @Effect()
  loadGames$ = this.actions$.ofType(gamesActions.LOAD_GAMES).pipe(
    switchMap(() => {
      return this.gamesService
        .getGames()
        .pipe(
          map(games => new gamesActions.LoadGamesSuccess(games)),
          catchError(error => of(new gamesActions.LoadGamesFail(error)))
        )
    })
  )
}
