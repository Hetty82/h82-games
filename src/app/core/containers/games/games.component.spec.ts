import { NO_ERRORS_SCHEMA } from '@angular/core'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'

import { GamesComponent } from './games.component'
import { GamesService } from '../../services'


describe('GamesComponent', () => {
  let component: GamesComponent
  let fixture: ComponentFixture<GamesComponent>

  beforeEach(async(() => {
    TestBed.configureCompiler({ preserveWhitespaces: false } as any).configureTestingModule({
      declarations: [ GamesComponent ],
      providers: [
        {
          provide: GamesService,
          useValue: {
            getGames: jest.fn()
          }
        }
      ],
      schemas: [ NO_ERRORS_SCHEMA ],
    })
    .compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(GamesComponent)
    component = fixture.componentInstance
    fixture.detectChanges()
  })

  it('should create component', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot()
  })
})
