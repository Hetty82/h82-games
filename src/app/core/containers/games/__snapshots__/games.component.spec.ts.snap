// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesComponent should match snapshot 1`] = `
<app-games
  games$="undefined"
  gamesService={[Function Object]}
>
  <h2>
    Game list
  </h2><ul>
    
  </ul>
</app-games>
`;
